### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by ori-sky**
This patch adds a pull request template in order to make it clearer what information should be included when opening pull requests with code contributions.

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+affects any compilation options or runtime flags, or changes expected
+input/output values, please document these effects here.*
+
+# Anything else we should know?
+
+* If there's any other relevant information we should know that may help us in understanding and verifying your patch, please include it here.
+
+# Checklist
+
+- [ ] I agree to follow the project [Code of Conduct](https://github.com/codeplaysoftware/oneapi-construction-kit/blob/main/CODE_OF_CONDUCT.md).
```
Comment (coldav) [inline pull_request_template.md:23]: I'm not sure we need a checklist. Is it possible to just add these as a comment?

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+effects and consequences of the change.*
+
+# Current behaviour
+
+*Describe the current behaviour of the project and how it differs from the
+behaviour your change is proposing.*
+
+# Expected behaviour
+
+*Describe the behaviour your change is proposing to the project. If your change
```
Comment (coldav) [inline pull_request_template.md:13]: These two fields blend into other - the current behaviour talks about how it differs - maybe that part should be dropped.
Also i think Expected is wrong - Describe the behaviour change may be a more accurate title.

File: pull_request_template.md (hunk 1 above)
Comment (ori-sky) [reply pull_request_template.md]: Thanks

File: pull_request_template.md (hunk 2 above)
Comment (ori-sky) [reply pull_request_template.md]: I've updated the sections now, does this look clearer?

**Review by coldav — APPROVED** [review]

**Pull request merged**